src/stateless/slider.vue

```diff
@@ -0,0 +1,546 @@
+            }
+        },
+    },
+    beforeCreate () {
+        this.ticksCount = 20
+        this.labelPadding = 6
+    },
+    created () {
+        if (this.decimalPlacesCount > 1)
+            throw new Error('Max one decimal place is supported.')
+
+        let isStepValid = (this.max - this.min) / this.step % 1 === 0
+        if (!isStepValid)
+            throw new Error('Difference between max and min is not divisible by step.')
+
+        let isValueValid = ((this.value - this.min) % this.step).toFixed(0) === '0'
+        if (!isValueValid)
+            throw new Error('Value is not a valid step.')
+
+        if (this.min < 0 || this.min > 998)
+            throw new Error('Min must be between 0 and 998.')
+
+        if (this.max > 999 || this.max <= this.min)
+            throw new Error('Max must be between min+1 and 999.')
+
+        if (this.value < this.min || this.value > this.max)
+            throw new Error('Value must be between min and max.')
+    },
+    mounted () {
+        this.isDomReady = true
+    },
+    methods: {
+        startDrag (e) {
+            if (this.disabled) return
+
+            this.isChanged = true
+            this.isDragging = true
+
+            this.setPosition(e)
+            window.addEventListener('mouseup', this.stopDrag)
+            window.addEventListener('mousemove', this.setPosition)
+            this.$refs.bar.focus()
+        },
+        setPosition (e) {
+            let percent = (e.clientX - this.sliderOffset) / this.sliderWidth
+            percent = Math.min(Math.max(0, percent), 0.999999)
+            let edgeStepOffset = this.stepPercentage / 2
+
+            let value = (Math.floor((percent - edgeStepOffset) / this.stepPercentage) + 1) * this.step + this.min
+
+            let roundingFactor = 1
+            let decimalsPart = this.step.toString().split('.')[1]
+            if (decimalsPart) {
+                roundingFactor = parseInt('1' + '0'.repeat(decimalsPart.length), 10)
+            }
+            let roundedValue = Math.round(value * roundingFactor) / roundingFactor
+
+            this.$emit('input', roundedValue)
+
+            e.preventDefault()
+        },
+        stopDrag () {
+            this.isDragging = false
+            window.removeEventListener('mouseup', this.stopDrag)
+            window.removeEventListener('mousemove', this.setPosition)
+        },
+        decreaseValue () {
+            if (!this.disabled && this.index > 0) {
+                this.isChanged = true
+                this.$emit('input', this.value - this.step)
+            }
+        },
+        increaseValue () {
+            if (!this.disabled && this.index < this.stepsCount) {
+                this.isChanged = true
+                this.$emit('input', this.value + this.step)
+            }
+        },
+        handleInput (value) {
+            let number = parseFloat(value)
+            this.$emit('input', isNaN(number) ? null : number)
```

Comment:
Since input already returns a number this method is probably not needed. You can just inline it `@input="$emit('input', $event)"`